pkg/service/service.go

```diff
@@ -479,20 +479,28 @@ func (s *Service) GetDeepCopyServices() []*lb.SVC {
 
 // RestoreServices restores services from BPF maps.
 //
+// It first restores all the service entries, followed by backend entries.
+// In the process, it deletes any duplicate backend entries that were leaked, and
+// are not referenced by any service entries.
+//
 // The method should be called once before establishing a connectivity
 // to kube-apiserver.
 func (s *Service) RestoreServices() error {
 	s.Lock()
 	defer s.Unlock()
+	var errs error
+	backendsById := make(map[lb.BackendID]struct{})
 
-	// Restore backend IDs
-	if err := s.restoreBackendsLocked(); err != nil {
-		return err
+	// Restore service cache from BPF maps
+	if err := s.restoreServicesLocked(backendsById); err != nil {
+		errs = multierr.Append(errs,
```

Comment:
Please add the missing import - 

`"go.uber.org/multierr"`

(This would go after `"github.com/sirupsen/logrus"`.)
Also, please run `go mod tidy && go mod vendor` and commit the changes.